packages/core/src/transactions/TransactionRequest.ts

```diff
@@ -1,20 +1,23 @@
+import { ethers } from 'ethers'
+
 export interface TransactionRequestData {
-  from: string
-  chainId: number
+  chainId?: number
   data?: string
-  gas: string
-  gasLimit: number
-  gasPrice: string
+  from: string
+  gas?: ethers.types.BigNumber
+  gasLimit?: ethers.types.BigNumber
+  gasPrice?: ethers.types.BigNumber
   to: string
   value?: string
 }
 
 export default class TransactionRequest {
-  readonly from!: string
   readonly chainId!: number
   readonly data?: string
-  readonly gasLimit!: number
-  readonly gasPrice!: string
+  readonly from!: string
+  readonly gas!: ethers.types.BigNumber
+  readonly gasLimit!: ethers.types.BigNumber
+  readonly gasPrice!: ethers.types.BigNumber
```

Comment:
Should we use strings here, to avoid exposing an object we don’t control? Ideally we should have `BigInt`s, which Ethers supports since v5, but I think Web3 is still having issues with it.